morph_tool/morphdb.py

```diff
@@ -0,0 +1,346 @@
+                 label: str = None):
+        '''MorphInfo ctor:
+
+        Upon initialization, all repair attributes are set to True.
+
+        Args:
+            name: the morphology name (without the extension)
+            mtype: the full mtype (ie 'L1_DAC:A')
+            layer: the layer (accepts both string and int)
+            label: (optional) a group label to be used to identify multiple groups of morphologies
+                added to the same collection.
+        '''
+        self.name = name
+        self.mtype = mtype
+        if self.MTYPE_SEPARATOR in mtype:
+            self.mtype_no_subtype, self.msubtype = mtype.split(self.MTYPE_SEPARATOR)
+        else:
+            self.mtype_no_subtype, self.msubtype = self.mtype, ''
+        self.layer = str(layer or '')
+
+        for attr in BOOLEAN_REPAIR_ATTRS:
+            setattr(self, attr, True)
+
+        self.axon_inputs = []
+        self.path = None
+        self.label = label
+
+    @classmethod
+    def _from_xml(cls, item: Dict[str, Any]):
+        '''MorphInfo ctor.
+        Args:
+            item: A dictionary that represents the content of the XML file.
+                  The only mandatory keys are [name, mtype, layer]
+        '''
+        morph = cls(
+            item['name'],
+            cls.MTYPE_SEPARATOR.join(filter(None, [item['mtype'], item.get('msubtype')])),
+            item['layer']
+        )
+
+        def is_true(el: Optional[str]) -> bool:
+            '''Parse a string representing a boolean repair flag and returns its boolean value
+            Unless clearly stated as false, missing tags default to True
+            - According to Eilif, an empty use_axon (corresponding to a null in the database)
+              means that the axon is supposed to be used
+            - dendrite_repair       defaults to True in BlueRepairSDK
+            - basal_dendrite_repair defaults to True in BlueRepairSDK
+            - unravel: well I guess we always want to do it
+            '''
+            assert el in {'true', 'false', 'True', 'False', None}, f'Invalid element: {el}'
+            return el in (None, '', 'true', 'True', )
+
+        repair = item.get('repair', {})
+        for attr in BOOLEAN_REPAIR_ATTRS:
+            setattr(morph, attr, is_true(repair.get(attr)))
+
+        axon_sources = repair.get('axon_sources')
+        morph.axon_inputs = axon_sources.get('axoninput', []) if axon_sources else []
+
+        # Case where there is a single <axoninput></axoninput> tag in the XML
+        if not isinstance(morph.axon_inputs, list):
+            morph.axon_inputs = [morph.axon_inputs]
+
+        return morph
+
+    @property
+    def row(self) -> List:
+        '''Flattened data structude ready to be used by a dataframe.'''
+        return [getattr(self, attr) for attr in COLUMNS]
+
+    def __repr__(self):
+        return f'MorphInfo(name={self.name}, mtype={self.mtype}, layer={self.layer})'
+
+
+def _create_dataframe(morphologies: List[MorphInfo]):
```

Comment:
Shouldn't this function be a `@staticmethod` of `MorphDB`?

Reply:
Hum, yes I could do this